src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaderBuilder.cs

```diff
@@ -89,10 +89,10 @@ public sealed class PEHeaderBuilder
             Subsystem = subsystem;
             DllCharacteristics = dllCharacteristics;
             ImageCharacteristics = imageCharacteristics;
-            SizeOfStackReserve = sizeOfStackReserve;
-            SizeOfStackCommit = sizeOfStackCommit;
-            SizeOfHeapReserve = sizeOfHeapReserve;
-            SizeOfHeapCommit = sizeOfHeapCommit;
+            SizeOfStackReserve = sizeOfStackReserve == 0 ? (Is32Bit ? 0x1000u : 0x4000u) : sizeOfStackReserve;
```

Comment:
This should stay as is. These stack sizes are not relevant for .NET Core. This is unnecessary behavior change that is potentially breaking.

Reply:
See the description

Reply:
I won't fix native AOT. The fix for native AOT would have to be in the native AOT build target files under https://github.com/dotnet/runtime/tree/main/src/coreclr/nativeaot/BuildIntegration

Reply:
Ok, I will have a look through there and fix it in my next commit 👍